### ISSUE TITLE
resource "cloudflare_rate_limit" has no `response.headers` field

### ISSUE THREAD

**Issue by grogancolin** [cloudflare/terraform-provider-cloudflare#196]
We cannot set `match.response.headers` in a `cloudflare_rate_limit`.
This is part of the Cloudflare UI to setting them, so I'd expect it to be here too.

### Terraform Version
 ```$ terraform -v
Terraform v0.11.11
+ provider.cloudflare v1.11.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- cloudflare_rate_limit

### Expected Behavior
That we can set `match.response.headers` in a `cloudflare_rate_limit`

### Actual Behavior
We cannot.

### References
The cloudflare-go library doesn't have this functionality either so would also need to be fixed there:
https://github.com/cloudflare/cloudflare-go/blob/master/rate_limiting.go#L38-L41

**Comment by jacobbednarz** [comment]
Thanks @grogancolin. I don't think this was around when I ported our rules into Terraform. Will add some upstream support shortly.

**Comment by jacobbednarz** [comment]
Thanks @grogancolin. I don't think this was around when I ported our rules into Terraform. Will add some upstream support shortly.

**Comment by jacobbednarz** [comment]
Add support to `cloudflare-go` via cloudflare/cloudflare-go#267. I'll add this to the Terraform provider now and we can leave it as pending until that has landed.

**Comment by grogancolin** [comment]
Heya @jacobbednarz - any update on getting this merged?
Looks like cloudflare/cloudflare-go#267 was merged a while ago now (maybe not released yet?)

Thanks!

**Comment by jacobbednarz** [comment]
@grogancolin Afraid I started on this and then got pulled onto a some other work. The change itself is a bit of a mess due to the structure of the response and needs some fiddling.

I'm working on a couple of other bits at the moment to enable us internally to manage some resources but still aiming to get back to this once they are done.

**Comment by mauricio** [comment]
Any news on when this will land?

**Comment by jacobbednarz** [comment]
Nothing from me as we still haven't needed it. PRs are welcome if this is something you want.

**Comment by vync** [comment]
Would like to open a PR for this but I believe I do not have the permissions to push my branch, could you please add me? 

Seeing:
```
ERROR: Permission to terraform-providers/terraform-provider-cloudflare.git denied to vync.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by jacobbednarz** [comment]
@vync You won't have permission to push changes to this repository (only maintainers do). You will need to create a fork and a pull request from that. Check out https://help.github.com/en/github/getting-started-with-github/fork-a-repo for some guidance on how to achieve it.

**Comment by vync-SB** [comment]
I was now taking another look at this and as it seems there is no problem with importing rate limiting rules anymore.

**Comment by danielmotaleite** [comment]
Sorry about the noise, but any update on this?

**Comment by jacobbednarz** [comment]
Nope, the linked PR stalled.

**Comment by jacobbednarz** [comment]
Closed via #911
